lib/common/common/src/iterator_ext/mod.rs

```diff
@@ -42,6 +46,68 @@ pub trait IteratorExt: Iterator {
     {
         OnFinalCount::new(self, f)
     }
+
+    /// Measures the hardware usage of an iterator.
+    ///
+    /// # Arguments
+    /// - `hw_acc`: accumulator holding a counter cell
+    /// - `multiplier`: multiplies the number of iterations by this factor.
+    /// - `f`: Closure to get the specific counter to increase from the cell inside the accumulator.
+    fn measure_hw_with_acc<R>(
```

Comment:
It doesn't feel like Hw related functions should be in the pretty low-level `IteratorExt`. I would prefer to have it on higher level of abstraction, similar to our discussion about MmapHashmap